src/ProductSync.php

```diff
@@ -679,6 +684,31 @@ public static function mark_feed_dirty( $product_id ) {
 		Pinterest_For_Woocommerce()::save_data( 'feed_dirty', true );
 	}
 
+	/**
```

Comment:
This is not really necessary.
`woocommerce_hide_out_of_stock_items` is passed through `update_option` function. That function already performs the old versus new check. If A change is detected the DB will be updated and action will be triggered:

"update_option_{$option}"

Without a change in value, this action will not be triggered. This means that we only need to listen to the action:

```PHP
			add_action(
				'update_option_woocommerce_hide_out_of_stock_items',
				function () {
					Pinterest_For_Woocommerce()::save_data( 'feed_dirty', true );
				}
			);
```

Reply:
Totally. Fixed.